AdobeStockImage/Model/AssetRepository.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockImage\Api\AssetRepositoryInterface;
+use Magento\AdobeStockImage\Api\Data\AssetInterface;
+use Magento\AdobeStockImage\Model\AssetFactory;
+use Magento\AdobeStockImage\Model\ResourceModel\Asset as ResourceModel;
+use Magento\AdobeStockImage\Model\ResourceModel\Asset\Collection as AssetCollection;
+use Magento\AdobeStockImage\Model\ResourceModel\Asset\CollectionFactory as AssetCollectionFactory;
+use Magento\Framework\Api\SearchCriteriaBuilderFactory;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\SearchResultsInterface;
+use Magento\Framework\Api\SortOrder;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class AssetRepository
+ * @package Magento\AdobeStockImage\Model
+ * @api
+ */
+class AssetRepository implements AssetRepositoryInterface
+{
+    /**
+     * @var ResourceModel
+     */
+    private $resource;
+
+    /**
+     * @var AssetFactory
+     */
+    private $factory;
+
+    /**
+     * @var AssetCollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * @var SearchResultsInterface
+     */
+    private $searchResult;
+
+    /**
+     * @var SearchCriteriaBuilderFactory
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * AssetRepository constructor.
+     * @param ResourceModel $resource
+     * @param AssetCollectionFactory $collectionFactory
+     * @param AssetFactory $factory
+     * @param SearchResultsInterface $searchResult
+     * @param SearchCriteriaBuilderFactory $searchCriteriaBuilderFactory
+     */
+    public function __construct(
+        ResourceModel $resource,
+        AssetCollectionFactory $collectionFactory,
+        AssetFactory $factory,
+        SearchResultsInterface $searchResult,
+        SearchCriteriaBuilderFactory $searchCriteriaBuilderFactory
+    ) {
+        $this->resource = $resource;
+        $this->collectionFactory = $collectionFactory;
+        $this->factory = $factory;
+        $this->searchResult = $searchResult;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilderFactory;
+    }
+
+    /**
+     * Save asset
+     * @param AssetInterface $item
+     * @return bool
+     */
+    public function save(AssetInterface $item) : bool
+    {
+        try {
+            $this->resource->save($item);
+            return true;
+        } catch (\Exception $ex) {
```

Comment:
I think it's better to leave the responsibility of catching exceptions to clients (so that they can retrieve exception details and report it accordingly).

Reply:
Got it. I changed that.